packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm

```diff
@@ -193,9 +193,10 @@ - (void)setBridge:(RCTBridge *)bridge
 
 - (void)setReaSurfacePresenter
 {
-  if (reaSurface == nil) {
-    // we need only one instance because SurfacePresenter is the same during the application lifetime
-    reaSurface = [[REAInitializerRCTFabricSurface alloc] init];
+  if(_surfacePresenter && ![_surfacePresenter surfaceForRootTag:reaSurface.rootTag]) {
+    if (!reaSurface) {
+      reaSurface = [[REAInitializerRCTFabricSurface alloc] init];
```

Comment:
Is it possible to change `reaSurface` from a static to a regular instance? Maybe that approach would be easier and safer.

Reply:
I'm not sure why was it made static in the first place, probably because `SurfacePresenter is the same during the application lifetime` which can be not true. I can change it then.